pcap/pcap.go

```diff
@@ -336,8 +379,27 @@ func (p *Handle) getNextBufPtrLocked(ci *gopacket.CaptureInfo) error {
 	}
 	var result NextError
 	for {
+		// test if we need to stop
+		select {
+		case <-p.stop:
+			return io.EOF
+		default:
+		}
+
+		if p.timeout < 0 {
+			C.pcap_wait(p.cptr, 0)
+		} else {
+			// need to wait less than the read timeout according to pcap
```

Comment:
According to http://www.tcpdump.org/manpages/pcap_get_selectable_fd.3pcap.html:

> select() and poll() do not work correctly on BPF devices; pcap_get_selectable_fd() will return a file descriptor on most of those versions (the exceptions being FreeBSD 4.3 and 4.4), but a simple select() or poll() will not indicate that the descriptor is readable until a full buffer's worth of packets is received, even if the read timeout expires before then. To work around this, an application that uses select() or poll() to wait for packets to arrive must put the pcap_t in non-blocking mode, and must arrange that the select() or poll() have a timeout less than or equal to the read timeout, and must try to read packets after that timeout expires, regardless of whether select() or poll() indicated that the file descriptor for the pcap_t is ready to be read or not. (That workaround will not work in FreeBSD 4.3 and later; however, in FreeBSD 4.6 and later, select() and poll() work correctly on BPF devices, so the workaround isn't necessary, although it does no harm.)

Does this mean that in FreeBSD 4.3-4.6, this code will fail to work?  Or that it will fail to work in FreeBSD 4.3 on?